libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,105 @@
+package com.mapbox.navigation.testing.ui
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import java.util.Date
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ */
+class MockLocationUpdatesRule : TestWatcher() {
```

Comment:
pretty cool that we can do this! if you get it all working, would be interested in putting it somewhere we can share it across repos